beacon-chain/rpc/prysm/v1alpha1/validator/proposer.go

```diff
@@ -142,6 +137,70 @@ func (vs *Server) GetBlock(ctx context.Context, req *ethpb.BlockRequest) (*ethpb
 	return blk, nil
 }
 
+func (vs *Server) depositAndPackAttestations(ctx context.Context, head state.BeaconState, eth1Data *eth.Eth1Data) ([]*eth.Deposit, []*eth.Attestation, error) {
+	if featureconfig.Get().EnableGetBlockOptimizations {
+		deposits, atts, err := vs.optimizedDepositAndPackAttestations(ctx, head, eth1Data)
+		if err != nil {
+			return nil, nil, err
+		}
+		return deposits, atts, nil
+	}
+
+	// Pack ETH1 deposits which have not been included in the beacon chain.
+	deposits, err := vs.deposits(ctx, head, eth1Data)
+	if err != nil {
+		return nil, nil, status.Errorf(codes.Internal, "Could not get ETH1 deposits: %v", err)
+	}
+
+	// Pack aggregated attestations which have not been included in the beacon chain.
+	atts, err := vs.packAttestations(ctx, head)
+	if err != nil {
+		return nil, nil, status.Errorf(codes.Internal, "Could not get attestations to pack into block: %v", err)
+	}
+
+	return deposits, atts, nil
+}
+
+func (vs *Server) optimizedDepositAndPackAttestations(ctx context.Context, head state.BeaconState, eth1Data *eth.Eth1Data) ([]*eth.Deposit, []*eth.Attestation, error) {
+	eg, egctx := errgroup.WithContext(ctx)
+	var deposits []*eth.Deposit
+	var atts []*eth.Attestation
+
+	eg.Go(func() error {
+		// Pack ETH1 deposits which have not been included in the beacon chain.
+		localDeposits, err := vs.deposits(ctx, head, eth1Data)
```

Comment:
Shouldn't you use egctx here?
```suggestion
		localDeposits, err := vs.deposits(egctx, head, eth1Data)
```